src/engine/config.cpp

```diff
@@ -23,11 +23,63 @@
 #include <fmt/printf.h>
 
 #ifdef _WIN32
+#include "winStuff.h"
 #define CONFIG_FILE "\\furnace.cfg"
 #else
+#include <unistd.h>
+#include <pwd.h>
+#include <sys/stat.h>
 #define CONFIG_FILE "/furnace.cfg"
 #endif
 
+void DivEngine::initConfDir() {
+#ifdef _WIN32
+  // maybe move this function in here instead?
+  configPath=getWinConfigPath();
+#elif defined (IS_MOBILE)
+  configPath=SDL_GetPrefPath();
+#else
+  // TODO this should check XDG_CONFIG_HOME first
+  char* home=getenv("HOME");
+  if (home==NULL) {
+    int uid=getuid();
+    struct passwd* entry=getpwuid(uid);
+    if (entry==NULL) {
+      logW("unable to determine home directory (%s)!",strerror(errno));
+      configPath=".";
+      return;
+    }
+  } else {
+    configPath=home;
+  }
+#  ifdef __APPLE__
+  configPath+="/Library/Application Support/Furnace";
+#  elif defined (__HAIKU__)
+  configPath+="/config/settings/furnace";
```

Comment:
on Haiku the recommended way to do this is to use find_directory() to locate B_USER_SETTINGS_DIRECTORY (and then append the application name to that)

https://www.haiku-os.org/legacy-docs/bebook/BDirectory.html#find_directory

Reply:
I'll give this a shot, although some non-legacy docs seem to suggest that it's "partially deprecated [except for] some use cases where it should be used"?

https://www.haiku-os.org/docs/develop/packages/Migration.html#changes-for-application-developers

Reply:
It has been "partially deprecated" in the sense that several of the available constants and directories were removed from the API. But the config directory is one of the cases where find_directory is still appropriate.

Reply:
Alright, should be done.

Reply:
Will do a new check. :)